pyspecdata/core.py

```diff
@@ -1623,43 +1618,32 @@ def concat(datalist, dimname, chop=False):
         )
     other_info_out = datalist[0].other_info
     if dimname in datalist[0].dimlabels:
-        print(
-            "The dimname you specified exists already in one of the datasets so I am assuming it also exists in the other"
+        assert all(
+            [dimname in datalist[j].dimlabels for j in range(len(datalist))]
+        ), (
+            "the dimension '"
+            + dimname
+            + "' exists in the first datset, but not the second!"
         )
         # {{{ make list of the data axes and dimname axes for datasets
         ndarrays = []  # will contain numpy array of data for each dataset
-        conc_axis = [] # will contain numpy array of the dimname axis for each dataset
+        conc_axis = (
+            []
+        )  # will contain numpy array of the dimname axis for each dataset
         for j in range(len(datalist)):
             ndarrays.append(datalist[j].data)
             conc_axis.append(datalist[j][dimname])
+        # PR: much more compact to just do this with a list comprehension in place, below
         # }}}
         # {{{ concatenate the data ndarrays and dimname axis ndarrays
-        newdata = np.concatenate(
+        concated_data = np.concatenate(
             tuple(ndarrays), axis=datalist[-1].dimlabels.index(dimname)
         )
-        new_dimname = np.concatenate(tuple(conc_axis))
-        # }}}
-        # {{{ Pull axis labels from the last item in the list
-        #     note this is similar to what is done below if the dimname
-        #     is not preexisting however here there is no newdimsize or
-        #     dimname so there is a slight difference
-        extra_dims = False
-        if len(datalist[-1].axis_coords) > 1:
-            extra_dims = True
-            extra_dimlabels = list(datalist[-1].dimlabels)
-            extra_axis_coords = list(datalist[-1].axis_coords)
-            dimname_idx = extra_dimlabels.index(dimname)
-            extra_dimlabels.pop(dimname_idx)
-            extra_axis_coords.pop(dimname_idx)
+        concated_ax_coords = np.concatenate(tuple(conc_axis))
         # }}}
-        newdatalist = nddata(newdata, datalist[0].dimlabels)
-        newdatalist.setaxis(dimname, new_dimname).set_units(
-            dimname, datalist[0].get_units(dimname)
-        )
-        newdatalist.sort(dimname)
-        if extra_dims:
-            # make sure we still have the indirect axes we started with
-            newdatalist.labels(extra_dimlabels[0], extra_axis_coords[0])
+        retval = datalist[-1].copy(data=False)
+        retval.axis_coords[datalist[-1].axn(dimname)] = concated_ax_coords
+        retval.data = concated_data
```

Comment:
note this is much simpler than above.  I think the stuff above also would not work if there were more than 2 dimensions.  **The internal attributes that store the coordinates, etc, rely very crucially on order -- dimlabels sets the order, and the corresponding element of, e.g. `axis_coords` is expected to contain the relevant attribute *for that dimension***